src/devices/CharacterLcd/Hd44780.cs

```diff
@@ -525,34 +526,29 @@ private void WriteBits(byte value, int count)
             // and 230ns on 5V. (PWeh on page 49/52 and Figure 25 on page 58)
 
             _controller.Write(_enablePin, PinValue.High);
-            DelayMicroseconds(1);
+            DelayMicroseconds(1, allowThreadYield: false);
             _controller.Write(_enablePin, PinValue.Low);
         }
 
-        private void DelayMicroseconds(int microseconds, bool checkBusy = true)
+        private void DelayMicroseconds(int microseconds, bool allowThreadYield = true,  bool checkBusy = true)
```

Comment:
Since there is only one call to this at the moment, is the `allowThreadYield` parameter necessary? If so would a default of `false` or no default (to require explicit specification by caller) be more appropriate?

Reply:
I'll make it explicit.

Reply:
Did you intend to make this change?

Reply:
Yes, there are multiple callers and only one. I made the helper explicit. I should have put this one at the end. To make this clearer I'm changing the name of the helper here to `WaitForNotBusy` and calling the new helpers directly when they aren't looking for not busy (in theory).